merrors/merrors.go

```diff
@@ -38,11 +39,42 @@ func (e *NilOrMultiError) Add(errs ...error) {
 }
 
 // Err returns the error list as an Error (also implements error) or nil if it is empty.
-func (e NilOrMultiError) Err() Error {
-	if len(e.errs) == 0 {
+func (e *NilOrMultiError) Err() Error {
+	if e == nil || len(e.errs) == 0 {
 		return nil
 	}
-	return multiError(e)
+
+	return multiError(*e)
+}
+
+// NilOrMultiSyncError is a thread-safe implementation of NilOrMultiError.
+// It allows combining multiple errors into one.
+type NilOrMultiSyncError struct {
```

Comment:
I wonder if we don't want to make the normal implementation concurrency safe 🤔 I don't think it introduces worth noting overhead... The only risk is when somebody would like to have different synchronization techniques that works for them, then they would need to reimplement all.. but sounds like rare use case. 

Otherwise another option is to have different package..

I think for me it would make more sense to have ONLY gorotoutine safe implementation here.

Reply:
I think it depends on what use cases there are out in the wild, but I'd agree having one implementation and documenting that it's concurrency safe so it's clear for the users is the way to go. I'll prepare the changes.